File: src/main/java/com/manson/fo76/domain/ItemsUploadFilters.java
package com.manson.fo76.domain;

import java.util.ArrayList;
import java.util.List;
import org.apache.commons.lang3.builder.ToStringBuilder;

public class ItemsUploadFilters {

	private List<Integer> filterFlags = new ArrayList<>();
	private boolean legendaryOnly;
	private boolean tradableOnly;

	public List<Integer> getFilterFlags() {
		return filterFlags;
	}

	public void setFilterFlags(List<Integer> filterFlags) {
		this.filterFlags = filterFlags;
	}

	public boolean isLegendaryOnly() {
		return legendaryOnly;
	}

	public void setLegendaryOnly(boolean legendaryOnly) {
		this.legendaryOnly = legendaryOnly;
	}

	public boolean isTradableOnly() {
		return tradableOnly;
	}

	public void setTradableOnly(boolean tradableOnly) {
		this.tradableOnly = tradableOnly;
	}

	@Override
	public String toString() {
		return new ToStringBuilder(this, org.apache.commons.lang3.builder.ToStringStyle.NO_CLASS_NAME_STYLE)
				.append("filterFlags", filterFlags)
				.append("legendaryOnly", legendaryOnly)
				.toString();
	}
}

File: src/main/java/com/manson/fo76/domain/ItemsUploadFilters.kt
package com.manson.fo76.domain

import org.apache.commons.lang3.builder.ToStringBuilder
import org.apache.commons.lang3.builder.ToStringStyle
import java.util.*

class ItemsUploadFilters {
    var filterFlags: List<Int> = ArrayList()
    var legendaryOnly = false
    var tradableOnly = false
    override fun toString(): String {
        return ToStringBuilder(this, ToStringStyle.NO_CLASS_NAME_STYLE)
                .append("filterFlags", filterFlags)
                .append("legendaryOnly", legendaryOnly)
                .toString()
    }
}